Reqnroll/Formatters/RuntimeSupport/FeatureLevelCucumberMessages.cs

```diff
@@ -1,21 +1,71 @@
 /// to be used at runtime.
 /// </summary>
-public class FeatureLevelCucumberMessages(Func<Source> source, Func<GherkinDocument> gherkinDocument, Func<IEnumerable<Pickle>> pickles)
+public class FeatureLevelCucumberMessages : IFeatureLevelCucumberMessages
 {
-    private Lazy<Source> _source = new Lazy<Source>(source);
-    private Lazy<GherkinDocument> _gherkinDocument = new Lazy<GherkinDocument>(gherkinDocument);
-    private Lazy<IEnumerable<Pickle>> _pickles = new Lazy<IEnumerable<Pickle>>(pickles);
-
-    public bool HasMessages => Source is not null && GherkinDocument is not null && Pickles is not null;
-    public Source Source { get => _source.Value; }
-    public GherkinDocument GherkinDocument { get => _gherkinDocument.Value; }
-    public IEnumerable<Pickle> Pickles { get => _pickles.Value; }
+    private Lazy<IEnumerable<Envelope>> _embeddedEnvelopes;
+    private readonly int _expectedEnvelopeCount;
+
+    public FeatureLevelCucumberMessages(string resourceNameOfEmbeddedMessages, int envelopeCount)
+    {
+        var assm = Assembly.GetCallingAssembly();
+        _embeddedEnvelopes = new Lazy<IEnumerable<Envelope>>(() => ReadEnvelopesFromAssembly(assm, resourceNameOfEmbeddedMessages));
+        _expectedEnvelopeCount = envelopeCount;
+    }
+
+    private IEnumerable<Envelope> ReadEnvelopesFromAssembly(Assembly assembly, string resourceNameOfEmbeddedNdJson)
+    {
+        var targetResourceName = resourceNameOfEmbeddedNdJson.Replace("/", "\\") + ".ndjson";
+
+        try
+        {
+            if (targetResourceName != null)
+            {
+                using (var stream = assembly.GetManifestResourceStream(targetResourceName))
+                {
+                    if (stream != null)
+                        using (var sr = new StreamReader(stream))
+                        {
+                            var content = sr.ReadToEnd();
+                            var lines = content.Split(new[] { "\r\n", "\n", "\r" }, StringSplitOptions.RemoveEmptyEntries);
+                            return lines.Select(l => NdjsonSerializer.Deserialize(l)).ToArray();
+                        }
+                }
+            }
+        }
+        catch (System.Exception)
+        {
+            return Enumerable.Empty<Envelope>();
+        }
+        return Enumerable.Empty<Envelope>();
+    }
+
+
+    public bool HasMessages
+    {
+        get
+        {
+            var hasSourceAndGD = Source is not null && GherkinDocument is not null;
+            var envelopeCount = hasSourceAndGD ? 
+                Pickles?.Count() + 2 
+                : 0;
+            return envelopeCount > 0 && envelopeCount == _expectedEnvelopeCount;
+        }
+    }
+
+    public Source Source => _embeddedEnvelopes.Value.Select(e => e.Source).FirstOrDefault(s => s != null);
+    public GherkinDocument GherkinDocument => _embeddedEnvelopes.Value.Select(e => e.GherkinDocument).FirstOrDefault(g => g != null);
```

Comment:
Is the GherkinDocument somewhere at the start of the _embeddedEnvelopes? Because if not then multiple calls of GherkinDocument (also via HasMessages) might be slow. Maybe we would need to put this to a `Lazy<>`

Same applies for `Source` and `Pickles` as they are also used within HasMessages as well.

Reply:
The generator, as it is currently implemented, puts the `Source` msg first, then the `GherkinDocument`, then the list of `Pickles`.
Would it be acceptable for this code to have in-built knowledge/assumptions about this order? If so, then these could be retrieved using indices (`Source = embeddedEnvelopers[0]`, etc).

Reply:
No, that's too much complication. I would probably wrap it with Lazy to be on the safe side.

Reply:
Substantial rewrite of that class, adding the lazy wrapper you requested. Please review again.